src/main/java/org/eclipse/yasson/internal/serializer/BooleanTypeDeserializer.java

```diff
@@ -41,7 +43,13 @@ public BooleanTypeDeserializer(Customization customization) {
 
     @Override
     public Boolean deserialize(JsonParser parser, DeserializationContext ctx, Type rtType) {
-        final JsonParser.Event event = ((JsonbParser) parser).moveToValue();
+        JsonValue current = parser.getValue();
```

Comment:
Instead of direct check for value is equal to boolean here, is it possible to refine [JsonbRiParser#moveTo(Event...)](https://github.com/eclipse-ee4j/yasson/blob/6ce6d8b91ff53ca1b2451ef993ba6e9cb9b6430d/src/main/java/org/eclipse/yasson/internal/JsonbRiParser.java#L193) method?

Reply:
That is what I first attempted to do when looking into a fix, but when deserializing a "naked" boolean the current `level` was always empty, and the code was requiring the opposite in order to peek the last event. I also didn't want to cause any additional performance overhead for the main path parser.
With this change, I've added some rather cheap operations which only impact the boolean deserializer.

If you want to take a crack at it feel free.  Unfortunately I couldn't think of a more elegant solution after staring at the code for a while.

Reply:
Looking more closely into a problem it is about inability of `javax.json.stream.JsonParser` to tell its current `JsonParser.Event`, because none has happened and we are not allowed to call next. And so a user has to _guess_ what a value should be and call appropriate getter - `getBoolean` in this case. I think a bug should be propagated to jsonp, since JsonParser should allow next() in this case returning `JsonParser.Event.VALUE_FALSE`. Your patch to save the day for Yasson is perfect than.